data/model-db/generate_historical_data.R

```diff
@@ -0,0 +1,778 @@
+  rename(run_id = id) |> 
+  select(-run, -description) |> 
+  # sex_id
+  left_join(sex, by = c("sex" = "definition")) |> 
+  rename(sex_id = id) |> 
+  select(-sex, -description) |> 
+  #direction_id
+  left_join(direction, by = c("passage_direction" = "definition")) |> 
+  rename(direction_id = id) |> 
+  select(-passage_direction, -description)
+
+try(if(any((unique(passage$survey_location_id) %in% survey_location$id) == F)) 
+  stop("Missing Survey Location ID! Please fix!"))
+try(if(any((unique(passage$run_id) %in% run$id) == F))
+  stop("Missing Run ID! Please fix!"))
+try(if(any((unique(passage$sex_id) %in% sex$id) == F))
+  stop("Missing Sex ID! Please fix!"))
+try(if(any((unique(passage$direction_id) %in% direction$id) == F))
+  stop("Missing Direction ID! Please fix!"))
+try(if(any(is.na(passage$date)))
+  stop("Missing Date! Please fix!"))
+  
+gcs_upload(passage,
+           object_function = f,
+           type = "csv",
+           name = "model-db/passage.csv",
+           predefinedAcl = "bucketLevel")
+# passage_estimates -------------------------------------------------------
+gcs_get_object(object_name = "standard-format-data/standard_adult_passage_estimate.csv",
+               bucket = gcs_get_global_bucket(),
+               saveToDisk = "data/standard-format-data/standard_adult_passage_estimate.csv",
+               overwrite = TRUE)
+passage_estimates_raw <- read_csv("data/standard-format-data/standard_adult_passage_estimate.csv")
+
+passage_estimate <- passage_estimates_raw |> 
+  # survey_location_id
+  mutate(reach = NA) |> 
+  left_join(survey_location, by = c("stream", "reach")) |>
+  select(-c(stream, reach, description)) |>
+  rename(survey_location_id = id) |>
+  # run_id
+  left_join(run, by = c("run" = "definition")) |> 
+  rename(run_id = id) |> 
+  select(-run, -description)   
+  
+try(if(any((unique(passage_estimate$survey_location_id) %in% survey_location$id) == F)) 
+  stop("Missing Survey Location ID! Please fix!"))
+try(if(any((unique(passage_estimate$run_id) %in% run$id) == F))
+  stop("Missing Run ID! Please fix!"))
+try(if(any(is.na(passage_estimate$year)))
+  stop("Missing Year! Please fix!"))
+
+gcs_upload(passage_estimate,
+           object_function = f,
+           type = "csv",
+           name = "model-db/passage_estimate.csv",
+           predefinedAcl = "bucketLevel")
+
+# daily_holding -----------------------------------------------------------------
+# TODO check run and adipose clipped
+
+gcs_get_object(object_name = "jpe-model-data/holding.csv",
+               bucket = gcs_get_global_bucket(),
+               saveToDisk = "data/model-data/holding.csv",
+               overwrite = TRUE)
+holding_raw <- read_csv("data/model-data/holding.csv")
+
+daily_holding <- holding_raw |> 
+  select(date, reach, count, latitude, longitude, stream) |> 
+  # remove all that are missing date
+  filter(!is.na(date)) |> 
+  mutate(run = "spring",
+         adipose_clipped = F) |> 
```

Comment:
Do we know that all ad_clip would be FALSE in this case? We may want to leave this column as NULL

Reply:
Is the thought that there may be some misidentified fall mixed in? I was assuming that since we don't have any Feather hatchery fish and that they wouldn't be adclipped but maybe we shouldnt assume there are no Feather hatchery?